lib/streamlit/watcher/local_sources_watcher.py

```diff
@@ -132,10 +149,20 @@ def _register_watcher(self, filepath, module_name):
             return
 
         try:
-            wm = WatchedModule(
-                watcher=PathWatcher(filepath, self.on_file_changed),
-                module_name=module_name,
-            )
+            if is_directory:
+                for root, _, files in os.walk(filepath):
```

Comment:
We do have logic that walks a full directory. https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/watcher/path_watcher.py#L143

It seems to just create a watch on the path, and that can allow for . https://github.com/streamlit/streamlit/blob/develop/lib/streamlit/watcher/path_watcher.py#L119

Perhaps we can just do a watch on the path itself. Perhaps we can modify `_register_watcher`. We can test if the glob pattern needs to be set. This would reduce watchers generated and perhaps simplify the logic.

Reply:
for sure i'll look at incorporating the `watch_dir` function and try to use the glob pattern to make watching paths easier